prog/main.go

```diff
@@ -134,6 +138,62 @@ type appFlags struct {
 	consulInf       string
 }
 
+type containerLabelFiltersPositive []app.APITopologyOption
+
+func (filters *containerLabelFiltersPositive) String() string {
+	return fmt.Sprint(*filters)
+}
+
+func (filters *containerLabelFiltersPositive) Set(value string) error {
+	*filters = append(*filters, makeAPITopologyOption(true, value, len(*filters)))
+	return nil
+}
+
+type containerLabelFiltersNegative []app.APITopologyOption
+
+func (filters *containerLabelFiltersNegative) String() string {
+	return fmt.Sprint(*filters)
+}
+
+func (filters *containerLabelFiltersNegative) Set(value string) error {
+	*filters = append(*filters, makeAPITopologyOption(false, value, len(*filters)))
+	return nil
+}
+
+func makeAPITopologyOption(positive bool, value string, filterNumber int) app.APITopologyOption {
```

Comment:
Sure, but you are not making an option from scratch you are translating a filter flag into it.